pkgs/top-level/aliases.nix

```diff
@@ -2,9 +2,21 @@ self:
 
 with self;
 
+let
+  # Removing recurseForDerivation prevents derivations of aliased attribute
+  # set to appear while listing all the packages available.
+  removeRecurseForDerivations = _n: alias: with lib;
+    if alias.recurseForDerivations or false then
+      removeAttrs alias ["recurseForDerivations"]
+    else alias;
```

Comment:
I thought of that, but I have not done it because this adds extra computations, which is not desirable.  For example if you have a large attribute set, you would decompose it to recompose it without one attribute.